man/fi_getinfo.3.md

```diff
@@ -512,6 +512,15 @@ input flags.  Valid flags include the following.
   the node and/or service parameter must be non-NULL.  This flag is
   often used with passive endpoints.
 
+*FI_PROV_ATTR_ONLY*
+: Indicates that the caller is only querying for what providers are
+  potentially available.  All providers will return exactly one
+  fi_info struct, regardless of whether that provider is usable on the
+  current platform or not.  The returned fi_info struct will contain
```

Comment:
I'm good with it. The providers will have to register themselves in ini() and postpone any driver calls, device checks etc. until getinfo() calls into them.